app/src/main/java/com/github/se/cyrcle/io/ProtoGenericSerializer.kt

```diff
@@ -0,0 +1,171 @@
+package com.github.se.cyrcle.io
+
+import androidx.datastore.core.Serializer
+import java.io.InputStream
+import java.io.OutputStream
+
+private const val END_OF_STRING_CHAR = 0.toChar()
+private const val END_OF_STREAM_CHAR = (-1).toChar()
+
+/**
+ * A class to easily serialize and deserialize class with basic variables inside. An example is
+ * available in the test folder, under Settings.
+ */
+abstract class ProtoGenericSerializer<T>(
+    override val defaultValue: T,
+) : Serializer<T> {
+
+  companion object {
+    private fun writeNByte(output: OutputStream, byteArray: ByteArray) {
```

Comment:
This is missing the plural (quite important since we write multiple byte*s* :) )